File: HUBSPOT_TO_S3/php/HubSpot.php
<?php

use HubSpot\Client\Crm\Contacts;
use HubSpot\Client\Crm\Companies;

class HubSpotAPI
{
    private HubSpot\Discovery\Discovery $hubspot;
    public function __construct($secret)
    {
        $apiKey = $secret['hubspot_basic_api_key'];
        $this->hubspot = HubSpot\Factory::createWithApiKey($apiKey);
    }

    /**
     * @throws Contacts\ApiException
     */
    public function create_contact($record)
    {
        $contactPayload = new Contacts\Model\SimplePublicObjectInput();
        $contactPayload->setProperties(array(
            'email' => $record['email'],
            'firstname' => $record['first_name'],
            'lastname' => $record['last_name'],
            'company' => $record['company'],
            'city' => $record['city'],
            'state' => $record['state'],
            'country' => $record['country'],
            'zip' => $record['zip']
        ));
        $this->hubspot->crm()->contacts()->basicApi()->create($contactPayload);
    }

    /**
     * @throws Companies\ApiException
     */
    public function create_company($record)
    {
        $companyPayload = new Companies\Model\SimplePublicObjectInput();
        $companyPayload->setProperties(array(
            'name' => $record['company'],
            'description' => $record['desc']
        ));
        $this->hubspot->crm()->companies()->basicApi()->create($companyPayload);
    }

    public function format_error($entity, $record, $e)
    {
        fwrite(
            STDERR,
            sprintf("ERR: Unable to create %s, %s: %s\n",
                $entity,
                var_export($record, true),
                var_export($e->getResponseBody(), true)
            )
        );
    }
}

File: HUBSPOT_TO_S3/ruby/hubspot.rb
require 'hubspot-api-client'

class HubSpot
    def initialize(secret, logger)
        @logger = logger
        Hubspot.configure do |configuration|
            configuration.api_key['hapikey'] = secret['hubspot_basic_api_key']
          end
    end

    def createContact(resource)
        properties = {
            "email": resource[0],
            "firstname": resource[1],
            "lastname": resource[2],
            "company": resource[3],
            "city": resource[4],
            "state": resource[5],
            "country": resource[6],
            "zip": resource[7]
          }
          simple_public_object_input = Hubspot::Crm::Contacts::SimplePublicObjectInput.new(properties: properties)
        begin
            api_response = Hubspot::Crm::Contacts::BasicApi.new.create(simple_public_object_input, auth_names: "hapikey")
            @logger.info("Contact created: #{resource[0]}")
        rescue Hubspot::Crm::Contacts::ApiError => e
            error_message = JSON.parse(e.response_body)['message']
            @logger.debug(error_message)
        end
    end

    def createCompany(resource)
        properties = {
            "name": resource[3],
            "city": resource[4],
            "state": resource[5],
            "country": resource[6],
            "zip": resource[7],
            "description": resource[8]
          }
          simple_public_object_input = Hubspot::Crm::Companies::SimplePublicObjectInput.new(properties: properties)
          begin
            api_response = Hubspot::Crm::Companies::BasicApi.new.create(simple_public_object_input, auth_names: "hapikey")
            @logger.info("Company created: #{resource[3]}")
          rescue Hubspot::Crm::Companies::ApiError => e
            error_message = JSON.parse(e.response_body)['message']
            @logger.debug(error_message)
          end
    end
end
